### DISCUSSION TITLE
Suggestion: week of month in due date

### DISCUSSION THREAD

**Discussion by ktprograms** [jotaen/xit#56] [File Format]
If I want to note that something should be done on the nth week of the month, currently I'd have to calculate which week it is in the year, then when reading it calculate back.

I propose adding support for a syntax like `2022/12/W1` (which complements the `2022/W48` syntax that currently exists).

**Comment by jotaen** [comment]
Thanks for sharing your thoughts. I see how that could be useful, but I think it wouldn’t be ISO-8601 compatible, so the spec would have to specify the rules around this on its own. I currently try to keep the spec slim and contain the overall complexity of it.

Working with dates by hand can always be slightly inconvenient to a certain degree (sometimes more, sometimes less – in your case maybe more). An alternative to altering the spec would be to ease this situation via tooling. For example, an editor plugin could provide inlined annotations/labels for working with dates, or offer extra functionality when typing, to take away some of the mental burden. As inspiration, the [Sublime Text package](https://github.com/jotaen/xit-sublime) has an autocomplete feature that if you type the token `2w`, it automatically computes the date “in two weeks from now”: (you can also type in `3w`, `5d`, `1m` etc.)

<img width="473" alt="Screenshot 2022-12-02 at 09 42 35" src="https://user-images.githubusercontent.com/3618384/205252608-9eb3dd77-8db0-4368-8d78-8e4a429e48fd.png">

<img width="370" alt="Screenshot 2022-12-02 at 09 44 56" src="https://user-images.githubusercontent.com/3618384/205252622-6622babb-f783-4e51-977a-9fc50a6754ba.png">

So maybe there is a way to achieve something similar for your use-case. The capabilities of course also depend on the editor engine.

**Reply by ktprograms** [reply]
Thanks for explaining. I think I'll just stick with whole month due dates, or add support for week of month in my own vim-xit fork. I feel like adding tooling for typing a looser date and then needing to interact with it in some way to display the looser date instead of the absolute date seems a bit too complicated, but that's just how I plan to use it (either very near due dates, or loose months or years).
